src/ddmd/backend/gflow.c

```diff
@@ -324,18 +324,44 @@ STATIC void accumrd(vec_t GEN,vec_t KILL,elem *n)
                 rdelem(&Gl,&Kl,n->E1);
                 rdelem(&Gr,&Kr,n->E2);
 
-                /* GEN = (GEN - Kl) | Gl |      */
-                /*      (GEN - Kr) | Gr         */
-                /* KILL |= Kl & Kr              */
+                switch (el_noreturn(n->E1) * 2 | el_noreturn(n->E2))
+                {
+                    case 0: // E1 and E2 return
+                        /* GEN = (GEN - Kl) | Gl |
+                         *       (GEN - Kr) | Gr
+                         * KILL |= Kl & Kr
+                         */
+                        vec_orass(Gl,Gr);
+                        vec_sub(Gr,GEN,Kl);
+                        vec_orass(Gl,Gr);
+                        vec_sub(Gr,GEN,Kr);
+                        vec_or(GEN,Gl,Gr);
+
+                        vec_andass(Kl,Kr);
+                        vec_orass(KILL,Kl);
```

Comment:
So you no longer trust Algebra ;)? Just adding a separate commit has the same effect btw, only that it avoids yet another TODO in master.